### DISCUSSION TITLE
Dynamic video duration with nextJS and lambda

### DISCUSSION THREAD

**Discussion by Wtiben** [remotion-dev/remotion#992] [Q&A] [answered]
Hi!

I'm using remotion in a nextJS application with `@remotion/player` on the frontend side to preview the video and @remotion/lambda on the server side to do the actual rendering. My video has a dynamic duration based on the sum duration of a couple of smaller videos which are passed through the inputProps. In order to achieve this with `@remotion/player`, i can simply calculate the duration of these individual videos by using `getVideoMetadata()`. This works fine.
I'm now trying to do the same thing on the server side and it's giving me some issues. `getVideoMetadata()` doesn't work here (error - ReferenceError: document is not defined) so it seems like the only simple way to set a dynamic duration is to do it in the rendering pipeline like in https://www.remotion.dev/docs/dynamic-metadata. My question is, would this not cause performance issues if every lamda instance has to calculate the video duration based on remote resources? And if so, do you have any idea if there is a better way to solve this?

**Comment by JonnyBurger** [comment, marked as answer]
Hi @Wtiben! Sorry for the slow response, I admit it slipped through the cracks!

The answer is that yes, there is a performance penalty since this code is executed in every chunk. Usually it is bearable but if the render needs to be highly optimized, you need to determine the duration of the video by other means. I usually recommend to use the FFMPEG library to do so using `ffmpeg -i [video-file]`.

This can then be passed into the Remotion render using input props https://www.remotion.dev/docs/parametrized-rendering#input-props to avoid having to fetch the asset in every chunk.
